js/ui/file_uploader.js

```diff
@@ -585,26 +577,26 @@ var FileUploader = Editor.inherit({
         file.isValidMaxSize = this._validateMaxFileSize(file);
     },
     _validateFileExtension: function(file) {
-        var allowedExtensions = this.option("validation.allowedFileExtensions");
+        var allowedExtensions = this.option("allowedFileExtensions");
         var fileExtension = file.value.name.substring(file.value.name.lastIndexOf('.'));
         if(allowedExtensions.length === 0) {
             return true;
         }
         for(var i = 0; i < allowedExtensions.length; i++) {
-            if(fileExtension === allowedExtensions[i]) {
+            if(fileExtension.toLowerCase() === allowedExtensions[i].toLowerCase()) {
```

Comment:
You call `fileExtension.toLowerCase()` each time. Maybe better get file extension in lower case early?